File: problems/748-shortest-completing-word/solution.py
class Solution:
    def shortestCompletingWord(self, licensePlate: str, words: List[str]) -> str:
        count = {}
        for char in licensePlate.lower():
            if 'a' <= char <= 'z':
                count[char] = count.get(char, 0) + 1
        
        res = ""
        for word in words:
            valid = True
            temp_count = count.copy()
            for char in word:
                if char in temp_count:
                    temp_count[char] -= 1
                    if temp_count[char] == 0:
                        del temp_count[char]
            
            if not temp_count:
                if not res or len(word) < len(res):
                    res = word
        return res

File: problems/748-shortest-completing-word/solution.cpp
class Solution {
public:
    string shortestCompletingWord(string licensePlate, vector<string>& words) {
        map<char, int> freq;
        for (char c : licensePlate) {
            c = tolower(c);
            if (isalpha(c)) {
                freq[c]++;
            }
        }
        string res = "";
        for (string word : words) {
            map<char, int> wordFreq;
            for (char c : word) {
                wordFreq[c]++;
            }
            bool complete = true;
            for (auto const& [key, val] : freq) {
                if (wordFreq.count(key) == 0 || wordFreq[key] < val) {
                    complete = false;
                    break;
                }
            }
            if (complete) {
                if (res == "" || word.length() < res.length()) {
                    res = word;
                }
            }
        }
        return res;
    }
};